solarforecastarbiter/reference_forecasts/tests/test_fx_utils.py

```diff
@@ -256,40 +281,104 @@ def test_get_data_start_end_labels_obs_avg_fx_instant(site_metadata):
         run_length=pd.Timedelta('1d'),
         interval_label='instant')
     with pytest.raises(ValueError) as excinfo:
-        utils.get_data_start_end(observation, forecast, run_time)
+        utils.get_data_start_end(observation, forecast, run_time, issue_time)
     assert 'made from interval average obs' in str(excinfo.value)
 
 
-@pytest.mark.parametrize("variable,expected_start,expected_end", [
+@pytest.mark.parametrize('rl,expected_start,expected_end,rt,lt', [
+    ('1d', '20190409T0000Z', '20190410T0000Z', '20190410T2230Z', '1h'),
+    ('1d', '20190409T0000Z', '20190410T0000Z', '20190410T0001Z', '1h'),
+    ('1d', '20190409T0000Z', '20190410T0000Z', '20190410T0001Z', '25h'),
+    ('1d', '20190409T0000Z', '20190410T0000Z', '20190410T0001Z', '49h'),
+    ('1d', '20190408T2300Z', '20190409T2300Z', '20190410T2230Z', '24h'),
+    # quirky that 23h ahead uses more recent data than 1h ahead, but
```

Comment:
It is quirky but I think I've convinced myself that it is correct. My understanding is:

`rt = '20190410T2230Z'`, `issue_time = '20190410T2300Z'`,  `lead_time_to_start = '23h'`, so fx from `20190411T2200Z` to `20190412T2200Z`. The most recent data I can use for that 24 hr period is `20190409T2200Z` to `20190410T2200Z`.

But if I set `rt = '20190410T2130Z'`, the most recent data I can use for that 24 hr period is `20190408T2200Z` to `20190409T2200Z`.

Reply:
or maybe I don't understand (see comment on "and that's ok" below)

Reply:
Again, more clear if I stated that the goal is to set the data period based on the forecast parameters and not based on when one tried to make a forecast. So the data period is always `20190409T2200Z` to `20190410T2200Z` for the `issue_time = '20190410T2300Z'` without regard for the run time. `run_persistence` has the safeguard so that it only produces the forecast for that issue time when `run_time > 20190410T2200Z`